apps/sentry/src/connectors/users.test.ts

```diff
@@ -1,65 +1,108 @@
-/* eslint-disable @typescript-eslint/no-unsafe-call -- test conveniency */
-/* eslint-disable @typescript-eslint/no-unsafe-return -- test conveniency */
-/**
- * DISCLAIMER:
- * The tests provided in this file are specifically designed for the `auth` connectors function.
- * Theses tests exists because the services & inngest functions using this connector mock it.
- * If you are using an SDK we suggest you to mock it not to implements calls using msw.
- * These file illustrate potential scenarios and methodologies relevant for SaaS integration.
- */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
 
+import { describe, test, expect, beforeEach } from 'vitest';
 import { http } from 'msw';
-import { describe, expect, test, beforeEach } from 'vitest';
 import { server } from '../../vitest/setup-msw-handlers';
-import { type MySaasUser, getUsers } from './users';
-import { MySaasError } from './commons/error';
+import { getUsers, type SentryUser, type Pagination, deleteUser } from './users';
+import { SentryError } from './commons/error';
 
-const validToken = 'token-1234';
-const maxPage = 3;
+const validToken = 'test-token';
+const sourceOrganizationId = 'test-organization';
+const cursor = 'test-cursor';
+const memberId= "test-member-id";
 
-const users: MySaasUser[] = Array.from({ length: 5 }, (_, i) => ({
-  id: `id-${i}`,
-  username: `username-${i}`,
-  email: `user-${i}@foo.bar`,
-}));
+const sentryUsers: SentryUser[] = [
+  {
+    role: 'admin',
+    user: {
+      id: 'user-id',
+      name: 'Username',
+      email: 'user@gmail.com',
+    },
+  },
+];
 
-describe('auth connector', () => {
-  describe('getUsers', () => {
-    // mock token API endpoint using msw
-    beforeEach(() => {
-      server.use(
-        http.get('https://mysaas.com/api/v1/users', ({ request }) => {
-          // briefly implement API endpoint behaviour
+const pagination: Pagination = {
+  nextCursor: 'next-test-cursor',
+};
+
+describe('getUsers', () => {
+  beforeEach(() => {
+    server.use(
+      http.get(`https://sentry.io/api/0/organizations/${sourceOrganizationId}/members/`, ({ request }) => {
+        if (request.headers.get('Authorization') !== `Bearer ${validToken}`) {
+          return new Response(undefined, { status: 401 });
+        }
+        const url = new URL(request.url); 
+        const lastCursor = 'last-cursor';
+        const nextCursor = 'next-cursor';
+        const requestCursor = url.searchParams.get('cursor');
+
+        if (requestCursor !== lastCursor) {
+          return new Response(
```

Comment:
create a single response and return headers conditionally(? statement) by checking requestCursor !== lastCursor.

Reply:
Done